mozci/push.py

```diff
@@ -1112,13 +1113,21 @@ def __make_label_summary_objects(pushes):
         [task for push in pushes for task in push.tasks], key=lambda x: x.label
     )
 
-    mapping = defaultdict(lambda: defaultdict(list))
-
     # Make a mapping keyed by task.label containing list of Task objects.
+    configs = {}
     for task in tasks:
-        mapping[task.label]["tasks"].append(task)
-
-    return [LabelSummary(key, value["tasks"]) for key, value in mapping.items()]
+        if task.configuration not in configs.keys():
+            # Dictionary to hold the mapping keyed by result.group mapped to list of
+            # task.id and list of and result.duration.
+            configs[task.configuration] = defaultdict(lambda: defaultdict(list))
+        configs[task.configuration][task.label]["tasks"].append(task)
```

Comment:
Nit: I know it was there before too, but why not dropping "tasks" and just doing:
```
configs[task.configuration][task.label].append(task)
```
so `configs[task.configuration] = defaultdict(lambda: defaultdict(list))` can become `configs[task.configuration] = defaultdict(list)`. Or `configs` could become `defaultdict(lambda: defaultdict(list))` instead of `{}` and you could remove the `if task.configuration not in configs.keys():` check.